lib/gradebook.rb

```diff
@@ -1 +1,52 @@
+class Gradebook
+  attr_reader :instructor, :courses, :all_students
+  def initialize (instructor)
+    @instructor = instructor
+    @courses = []
+    @all_students_a
+  end
 
+  def add_course(course)
+    @courses << course
+  end
+
+  def all_students
+    all_students = []
+    @courses.each {|course| all_students << course.students}
+    all_students.flatten
+  end
+
+  def list_all_students
+    all_students_h = Hash.new {|hash, key| hash[key] = []}
```

Comment:
You could use this more succinct syntax to create a hash with default values:
```suggestion
    all_students_h = Hash.new([])
```